src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -131,10 +131,12 @@ class IOUAmountPage extends React.Component {
             return;
         }
 
-        this.setState((prevState) => {
-            const amount = `${prevState.amount}${key}`;
-            return this.validateAmount(amount) ? {amount: this.stripCommaFromAmount(amount)} : prevState;
-        });
+        if (this.state.amount.length < CONST.IOU.AMOUNT_MAX_LENGTH) {
```

Comment:
Wouldn't this make the effective max length `13`?

Reply:
If the current length is < 14 then allow updating the state, else skip it. I tested this. But wouldn't matter as based on @marcaaron's comment I'll be updating the condition and the early return.